contrib/misc/statefile-parser/statefile.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python3
+
+#   BAREOS® - Backup Archiving REcovery Open Sourced
+#
+#   Copyright (C) 2024-2024 Bareos GmbH & Co. KG
+#
+#   This program is Free Software; you can redistribute it and/or
+#   modify it under the terms of version three of the GNU Affero General Public
+#   License as published by the Free Software Foundation and included
+#   in the file LICENSE.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+"""This module parses a Bareos filedaemon statefile.
+
+Passing the raw data of the statefile to parse_statefile() will return a list
+of JobResult objects. A code sample can be found is at the end of this file.
+
+This module will probably not work for files generated with the windows version
+of Bareos, as structs are packed differently.
+"""
+
+from glob import glob
+from struct import unpack, calcsize
+from dataclasses import dataclass
+from datetime import datetime
+
+HEADER_FMT = "".join(
```

Comment:
I think giving the specifiers useful names makes this much clearer.  What do you think ?

```python
def padding(count):
    return str(count) + "x"

def cstring(max_size):
    return str(max_size) + "s"

def repeat(type, count):
    return str(count) + type

i32 = "i"
i64 = "q"

u32 = "I"
u64 = "Q"

HEADER_FMT = "".join(
    (
        "=",             # native byte-order with standard sizes
	cstring(14),     # magic
	padding(2),
	i32,             # version
        padding(4),  
	u64,             # last job offset
	u64,             # end of list
	repeat(u64, 19), # reserved
    )
)
HEADER_LEN = calcsize(HEADER_FMT)

NUM_FMT = "".join(
    (
        "=", # native byte-order with standard sizes
        i32, # number of result records
    )
)
NUM_FMT_LEN = calcsize(NUM_FMT)

RESULT_FMT = "".join(
    (
        "=",           # native byte-order with standard sizes
	padding(16),
        i32,           # errors
        i32,           # job type
        i32,           # job status
        i32,           # job level
        u32,           # job id
        u32,           # volume session id
        u32,           # volume session time
        u32,           # job files
        u64,           # job bytes
        u64,           # job start time
        u64,           # job end time
        cstring(128),  # job name
    )
)
RESULT_LEN = calcsize(RESULT_FMT)
```

Reply:
ofc `job [start|end] time` should be `i64` here as well.

Reply:
I don't say that this is wrong. However, you will have to understand `struct.pack()`/`struct.unpack()` anyway. So I'm not convinced an extra level of indirection will really improve readability.

Reply:
Makes sense.